src/sysrepo.c

```diff
@@ -352,8 +352,22 @@ sr_connect(const sr_conn_options_t opts, sr_conn_ctx_t **conn_p)
         sr_conn_free(conn);
         if (created) {
             /* remove any created SHM so it is not considered properly created */
-            shm_unlink(SR_MAIN_SHM);
-            shm_unlink(SR_EXT_SHM);
+            sr_error_info_t *tmp_err = NULL;
+            char *shm_name = NULL;
+            if ((tmp_err = sr_path_main_shm(&shm_name))) {
+                sr_errinfo_merge(&err_info, tmp_err);
+            } else {
+                shm_unlink(shm_name);
+                free(shm_name);
+                shm_name = NULL;
```

Comment:
Seems that this line is not needed.

Reply:
What line exactly? If `sr_path_main_shm` failed, `shm_name` is `NULL`.

Reply:
But it does not matter whether `shm_name` is NULL or not in the code. I would just like to keep this code short because it is only error handling.

Reply:
I don't understand what I need to change here.

If `shm_name` is `NULL`, `shm_unlink` should not be called. If it is not `NULL`, it must be freed after calling `shm_unlink`. Did you have something else in mind?

Reply:
We seem not to understand each other, all I am saying is that the code `shm_name = NULL;` seems redundant to me and can be removed. If you want to keep it consistent then I would make sure the functions either allocate `shm_name` on success or set it to `NULL` on error. Maybe even `shm_name` initialization can be removed but since it is not taking up another line, I leave that up to you.

Reply:
We did not understand each other, indeed :) I'll remove that.